libtelemetry/src/main/java/com/mapbox/android/telemetry/NavigationMetadata.java

```diff
@@ -45,17 +45,25 @@
   private Boolean batteryPluggedIn;
   private Integer batteryLevel;
   private String connectivity;
+  private String tripIdentifier;
+  private Integer legIndex;
+  private Integer legCount;
+  private Integer stepIndex;
+  private Integer voiceIndex = null;
```

Comment:
What are benefits of having this value initialized to null?

Reply:
This is helpful for the null checks when serializing the data lower in the class. We want to know what to write in the bytes.

Reply:
i'd be interested to evaluate an option of replacing boxed types with Optional<T> to model optional parameters.